command_server.sh

```diff
@@ -122,6 +122,22 @@ $1
 	echo "{\"Done\": true, \"Exit\": $__cmd_last_status, \"Dir\": \"$dir\"}"
 }
 
+__cmd_complete() {
+    prog="${1%% *}"
+    if [[ "${#1}" -gt "${#prog}" ]]; then
+        # argument parsing
+        comp_command="$(complete -p "$prog" | sed s/^complete/compgen/)"
+        if [[ -z "$comp_command" ]]; then
+            echo ""
+        else
+            #echo $comp_command
+            echo "${comp_command% *} \"${1}\""
+            __cmd_run "${comp_command% *} \"${1}\""
+        fi
+    else
+        __cmd_run "compgen -abc $prog | sort -u"
```

Comment:
There are some complications if you try to use __cmd_run here. __cmd_last_status will be overwritten by the status of the compgen command, and history entries will be created.

I think a better option is to add your own (JSON) framing around the compgen output. (Something like `{Complete: "..."}`) Write it directly to stdout here and look for those messages in the asynchronous read loop in bash.go. The only tricky part is escaping the output properly.